docs/aws/audit/codebuildmonitoring/rules/codebuild_project_source_repo_url_remediation.mdx

```diff
@@ -1,7 +1,240 @@
-### Remediation
 
-#### Using Console
+### Triage and Remediation
+<Tabs>
+
+
+<Tab title='Prevention'>
+### How to Prevent
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+To prevent Sign-in Credentials from being included in Bitbucket Source Repository URLs in AWS CodeBuild using the AWS Management Console, follow these steps:
+
+1. **Review and Update Environment Variables:**
+   - Navigate to the AWS CodeBuild console.
+   - Select the specific build project you want to configure.
+   - Go to the "Build details" section and click on "Edit."
+   - Under the "Environment" section, ensure that sensitive information such as credentials is not hardcoded in the environment variables. Use AWS Secrets Manager or AWS Systems Manager Parameter Store to securely manage sensitive data.
+
+2. **Use IAM Roles for Authentication:**
+   - Ensure that your CodeBuild project is using an appropriate IAM role with the necessary permissions to access Bitbucket.
+   - Navigate to the "Service role" section in your CodeBuild project settings.
+   - Attach a policy to the IAM role that grants the necessary permissions without embedding credentials in the source repository URL.
+
+3. **Configure Source Repository URL Securely:**
+   - In the "Source" section of your CodeBuild project, ensure that the Bitbucket repository URL does not contain any embedded credentials.
+   - Use HTTPS URLs without credentials and configure authentication through OAuth or personal access tokens stored securely in AWS Secrets Manager.
+
+4. **Enable Logging and Monitoring:**
+   - Enable AWS CloudTrail and AWS Config to monitor changes and access to your CodeBuild projects.
+   - Set up alerts for any unauthorized changes or access patterns that might indicate credentials being exposed or misused.
+
+By following these steps, you can ensure that sign-in credentials are not included in Bitbucket source repository URLs in AWS CodeBuild, thereby enhancing the security of your build processes.
+</Accordion>
+
+<Accordion title='Using CLI'>
+To prevent the inclusion of sign-in credentials in Bitbucket source repository URLs in AWS CodeBuild using the AWS CLI, follow these steps:
+
+1. **Create an IAM Role with Least Privilege**:
+   Ensure that the IAM role used by CodeBuild has the least privilege necessary to perform its tasks. This minimizes the risk of exposing sensitive information.
+
+   ```sh
+   aws iam create-role --role-name CodeBuildServiceRole --assume-role-policy-document file://trust-policy.json
+   aws iam attach-role-policy --role-name CodeBuildServiceRole --policy-arn arn:aws:iam::aws:policy/AWSCodeBuildDeveloperAccess
+   ```
+
+2. **Use AWS Secrets Manager for Storing Credentials**:
+   Store your Bitbucket credentials in AWS Secrets Manager instead of hardcoding them in your source repository URL.
+
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>